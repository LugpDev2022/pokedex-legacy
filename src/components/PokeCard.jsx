import PropTypes from 'prop-types';

export const PokeCard = ({ pokemon }) => (
  <div className='col-12 col-sm-6 col-lg-3 mt-4'>
    <div className='card card-img-top'>
      <img src={sprite} alt='Pokemon Image' />
      <div className='card-body shadow'>
        <h4 className='card-title text-capitalize'>{name}</h4>
        <span className='card-text'>Index: {id}</span>
      </div>
    </div>
  </div>
);

PokeCard.propTypes = {
  pokemon: PropTypes.object.isRequired,
};
