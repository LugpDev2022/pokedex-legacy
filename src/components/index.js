export * from './Navbar';
export * from './PokeCard';
