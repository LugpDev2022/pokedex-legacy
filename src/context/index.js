export * from './PokemonContext';
export * from './PokemonProvider';
export * from './pokemonReducer';
export * from './types/types';
