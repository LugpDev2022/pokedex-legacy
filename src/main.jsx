import React from 'react';
import ReactDOM from 'react-dom/client';

import * as bootstrap from 'bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

import PokedexApp from './PokedexApp';
import { BrowserRouter } from 'react-router-dom';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <PokedexApp />
    </BrowserRouter>
  </React.StrictMode>
);
