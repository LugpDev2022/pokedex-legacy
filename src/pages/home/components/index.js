export * from './PokeCardGrid';
export * from './ChangePageButton';
