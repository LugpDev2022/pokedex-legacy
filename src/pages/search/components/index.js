export * from './SearchForm';
export * from './ShowCards';
