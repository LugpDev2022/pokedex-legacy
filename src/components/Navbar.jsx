import { useContext } from 'react';

import { Link } from 'react-router-dom';
import { BiUser, BiLogIn } from 'react-icons/bi';
import { FaSearch } from 'react-icons/fa';

import { PokemonContext } from '../context';

export const Navbar = () => {
  const { logged } = useContext(PokemonContext);

  return (
    <div
      className='navbar navbar-expand-md navbar-light'
      style={{ backgroundColor: '#5986fd' }}
    >
      <div className='container justify-content-between'>
        <Link to='/' className='navbar-brand d-flex align-items-center gap-3'>
          <h1 className='mb-0 pokemonFont'>Pokedex</h1>
        </Link>
        <button
          className='navbar-toggler'
          type='button'
          data-bs-toggle='collapse'
          data-bs-target='#navbarContent'
          aria-controls='navbarContent'
          aria-expanded='false'
          aria-label='Toggle navigation'
        >
          <span className='navbar-toggler-icon'></span>
        </button>

        <div
          className='collapse navbar-collapse justify-content-end'
          id='navbarContent'
        >
          <div className='navbar-nav d-flex align-items-center gap-4'>
            <Link to='/search' className='text-dark text-decoration-none'>
              <FaSearch className='fs-3 d-none d-md-block' />

              <span className='btn btn-outline-dark mt-3 d-block d-md-none'>
                Search a Pok√©mon
              </span>
            </Link>

            {logged ? (
              <button className='btn btn-outline-dark d-flex align-items-center gap-2'>
                <BiUser className='fs-2' />
                <span className='fs-5'>Account</span>
              </button>
            ) : (
              <button className='btn btn-outline-dark d-flex align-items-center gap-2'>
                <BiLogIn className='fs-3' />
                <span className='fs-5'>Log In</span>
              </button>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
