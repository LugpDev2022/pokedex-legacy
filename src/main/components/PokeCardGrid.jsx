import { PokeCard } from '../../components';
import { usePokemon } from '../hooks';

export const PokeCardGrid = () => {
  const { pokemons } = usePokemon();

  //TODO: Show spinner while chargins
  return (
    <div className='container px-5 mt-2'>
      <div className='row justify-content-between'>
        {pokemons.map(pokemon => {
          return <PokeCard key={pokemon.name} pokemon={pokemon} />;
        })}
      </div>
    </div>
  );
};
