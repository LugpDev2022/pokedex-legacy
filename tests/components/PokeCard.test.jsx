import { render, screen } from '@testing-library/react';
import { PokeCard } from '../../src/components';

describe('Tests in <PokeCard />', () => {
  const pokemon = {
    name: 'Chikorita',
    sprites: {
      front_default: 'https://chikorita.com/',
    },
    id: '57',
  };
  test('should match with snapshot', () => {
    const { container } = render(<PokeCard pokemon={pokemon} />);

    expect(container).toMatchSnapshot();
  });

  test('should show the pokemon name', () => {
    render(<PokeCard pokemon={pokemon} />);

    expect(screen.getByText(pokemon.name));
  });

  test('should show the given image', () => {
    render(<PokeCard pokemon={pokemon} />);
    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(pokemon.sprites.front_default);
    expect(alt).toBeTruthy();
  });
});
