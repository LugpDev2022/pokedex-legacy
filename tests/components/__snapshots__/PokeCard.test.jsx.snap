// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PokeCard /> should match with snapshot 1`] = `
<div>
  <div
    class="col-12 col-sm-6 col-lg-3 mt-4"
  >
    <div
      class="card card-img-top"
    >
      <img
        alt="Pokemon Image"
        src="https://chikorita.com/"
      />
      <div
        class="card-body shadow"
      >
        <h4
          class="card-title text-capitalize"
        >
          Chikorita
        </h4>
        <span
          class="card-text"
        >
          Index: 
          57
        </span>
      </div>
    </div>
  </div>
</div>
`;
